Tests/PackageDescriptionTests/VersionTests.swift

```diff
@@ -22,6 +22,13 @@ class VersionTests: XCTestCase {
         XCTAssertLessThan(Version("1.2.3-alpha.beta.2"), Version("1.2.3-alpha.beta.3"))
 
         XCTAssertEqual(Version("1.2.3-alpha.beta.2").description, "1.2.3-alpha.beta.2")
+        
+        
+        
+        XCTAssertNotNil(Version("0.0.0"))
+        XCTAssertNotNil(Version("1.2.3"))
+        XCTAssertNil(Version("4.2"))
```

Comment:
There is a pair of overloaded initialisers: `Version.init(_ versionString: String)` and `Version.init(_ version: Version)` defined in `PackageDescription/Version+StringLiteralConvertible.swift`.

My intention here is to test `Version.init(_ versionString: String)`. However, the compiler resolves this `Version(_: "4.2")` as a call to `Version.init(_: Version.init(stringLiteral: "4.2"))`, because `Version` conforms to `LosslessStringConvertible`. 

I attempted to temporarily work around it by removing the declaration of `Version.init(_ version: Version)`, but strangely it didn't work. The compiler keeps resolving the call to `Version.init(_ version: Version)` even when the declaration is removed. I tried removing `DerivedData/` and recompiled the project for multiple times, but somehow `Version.init(_ version: Version)` is stuck there.

Removing `Version.init(_ version: Version)` probably can't be a permanent solution anyway, because it would be a source-breaking change.

Reply:
```swift
var v4_2 = "4.2"
XCTAssertNil(Version(v4_2))
```

Reply:
🤦‍♂️ This is so obvious, and somehow I didn't think of it.

Thanks!